app/models/miq_worker.rb

```diff
@@ -514,7 +514,7 @@ def minimal_class_name
 
   def database_application_name
     zone = MiqServer.my_server.zone
-    "MIQ #{Process.pid} #{minimal_class_name}[#{compressed_id}], s[#{miq_server.compressed_id}], #{zone.name}[#{zone.compressed_id}]".truncate(64)
+    "MIQ|#{Process.pid}|#{miq_server.compressed_id}|#{compressed_id}|#{zone.compressed_id}|#{minimal_class_name}|#{zone.name}".truncate(64)
```

Comment:
I think it's fairly easy to assert that a length of 10 is near the maximum for each of the compressed ids:

```
irb(main):016:0> ApplicationRecord.compress_id(199000000500201)
=> "199r500201"
irb(main):017:0> ApplicationRecord.compress_id(199000000500201).length
=> 10
```

So, 

```
irb(main):012:0> "MIQ|12345|199r500201|199r500201|199r500201|".length
=> 43
```

leaves 20 characters for the class name and zone name (one is taken by `|`).  It's possible that the class name is longer than 20 characters and you lose the zone name entirely.